### ISSUE TITLE
ObjectID renders to an empty object when toArray is called on a result

### ISSUE THREAD

**Issue by fmestrone** [mongodb/mongo-php-driver#239]
Given the simple example here

```
    $client = new MongoDB\Client;
    $collection = $client->kanjisensei->KanjiCourses;
    $result = $collection->find([],['projection' => ['lessons' => false]]);
    error_log(json_encode($result->toArray()));
```

That's what is printed

```
[{"_id":{},"name":"Intermediate Kanji Book: Volume 1"},{"_id":{},"name":"Intermediate Kanji Book: Volume 2"}]
```

I wonder whether this is a problem with json_encode()? Is there a workaround? Or should I just iterate over each element and manually get the OID out of each ID object?

Thanks

**Comment by jmikola** [comment]
This is due to the BSON types (`MongoDB\BSON\ObjectID` one of them) not implementing JsonSerializable. That feature request is being tracked in [PHPC-459](https://jira.mongodb.org/browse/PHPC-459) and should be included in 1.2.0.

If you'd like the ObjectID rendered as [extended JSON](https://docs.mongodb.org/manual/reference/mongodb-extended-json/#data_oid), I would suggest filtering the array of documents through `array_map()` and changing the `_id` field from an ObjectID instance to `[ '$oid' => (string) $document['_id'] ]`. Alternatively, you may just want to cast the ObjectID to a string and return its hexadecimal value.

**Comment by fmestrone** [comment]
Great, thanks! I have solved by iterating through results and extracting data at each loop for now, but will keep an eye on PHPC-459.
